Dockerfile

```diff
@@ -1,21 +1,24 @@
 # DOCKER-VERSION        1.3.2
 
-# Build UI assets
+# Build UI assets stage
 FROM node:8-alpine as assets
 RUN apk add git
 RUN npm install -g gulp-cli
 RUN mkdir -p /work
 COPY bootstrap /work/
 WORKDIR /work
-RUN npm install && npm rebuild node-sass && gulp
+RUN npm install && npm rebuild node-sass && gulp sass
```

Comment:
Let's leave the `sass` target out, please; we may want to build more assets in the future, and I'd rather not have to hunt down why the assets are not present in the container.

Reply:
If I remove it the image will not be created.

When I run: docker-compose up

[09:25:33] Using gulpfile /work/gulpfile.js
[09:25:33] Starting 'default'...
[09:25:33] Starting 'sass'...
[09:25:35] Finished 'sass' after 2.02 s
[09:25:35] Starting '<anonymous>'...

Then it waits and the script doesn't continue.

OS: Mac OS
Docker version 18.09.2
docker-compose version 1.23.2

Reply:
I understand this change now... but don't know why it worked previously.

Basically, the `default` gulp target fires up `gulp.watch`, which is why that build step doesn't end for you.

Interestingly, in Linux, `gulp.watch` appears to detect it's in an environment that cannot spawn a browser, and it thus exits.

I think to solve this, I'm going to create a "deploy" target in the `gulpfile` where we can add multiple targets; that way, if we do build JS or other assets later, we can stuff them in there, and not need to change the `Dockerfile`.